numpy/core/src/multiarray/methods.c

```diff
@@ -2756,6 +2756,13 @@ NPY_NO_EXPORT PyMethodDef array_methods[] = {
         (PyCFunction) array_format,
         METH_VARARGS, NULL},
 
+    /* for typing; requires python >= 3.9 */
+    #ifdef Py_GENERICALIASOBJECT_H
+    {"__class_getitem__",
+        (PyCFunction)Py_GenericAlias,
+        METH_CLASS | METH_O, NULL},
+    #endif
+
```

Comment:
As `Py_GenericAlias` is only available for python >= 3.9, there are two ways of dealing with this limitation:
* Conditionally define the `__class_getitem__` method (as is currently done).
* Always define `__class_getitem__`, but let it raise for python 3.8.

I'm personally not a huge fan of methods that always raise, hence the first approach (this is open for discussion though).